grails-app/views/inventoryItem/_showStockHistory.gsp

```diff
@@ -284,9 +284,12 @@
                             title="${stockHistoryEntry?.inventoryItem?.lotStatus == LotStatusCode.RECALLED ? warehouse.message(code: 'inventoryItem.recalledLot.label') : ''}">
                             <td  class="middle">
                                 <g:if test="${stockHistoryEntry?.showDetails}">
-                                    <g:if test="${stockHistoryEntry?.transaction?.transactionType?.transactionCode== TransactionCode.DEBIT}">
-                                        <img src="${createLinkTo(dir: 'images/icons/silk', file: 'delete.png' )}" title="${format.metadata(obj:stockHistoryEntry?.transaction?.transactionType)}"/>
+                                    <g:if test="${stockHistoryEntry?.isInternal}">
+                                        <img src="${createLinkTo(dir: 'images/icons/silk', file: 'arrow_refresh_small.png' )}" title="Internal"/>
```

Comment:
Sorry I just commented on the specific commit instead of the PR. We can't add another transaction code. The system only supports the three that are there. This feels like something that should live in the level above transaction code ... maybe in the transaction or transaction type. If you're just trying to find a place for the localization key ... you could create a new classification (enum) for internal vs external.

Reply:
@jmiranda could we make it as:
```
enum TransactionSource {
    INTERNAL,
    EXTERNAL
}
```
Or we should avoid using the `Transaction` word in the naming at all?

Reply:
But you're correct that it might be described better by something other than Transaction.

Reply:
TransactionScope?

Reply:
TransactionTypeScope or TransactionTypeClass? I dislike "Class" because it's a keyword, so perhaps TransactionTypeCategory or something similar would be better.

Reply:
Other words that popped into my head are "operation" or "process" but I think those would replace transaction and not the thing that we're trying to think of.

So here are the words we get to choose from  
* Type
* Class
* Scope
* Boundary
* Context

And here are the descriptors that the above word is described by in case that helps to spawn more thoughts on the right word:
* Internal vs External
* Local vs Global
* Intra vs Extra 

Also ... https://martinfowler.com/bliki/TwoHardThings.html

Reply:
I like the Scope the most.
```
enum TransactionTypeScope {
    INTERNAL,
    EXTERNAL
}
```
@DaRoTP let's go with this version.

Reply:
I changed my mind on tying it to TransactionType. I think it makes more sense to call it TransactionScope since it describes the scope of the transaction and it might actually be derived by information on the transaction right? i.e. An INTERNAL transaction is one where origin == destination and an EXTERNAL transaction is one where the origin != destination.

Reply:
Therefore isInternal and isExternal should be methods on the Transaction, right? 

Another thought I had was that you could preprocess the transaction entries to keep only one of the transactions of an internal transaction pair. 

Note: LocalTransfer provides us with a link between two transactions but Local in this case means "managed by OB" not "internal". In other words, you could have a LocalTransfer that is external (i.e. debit transaction from a depot and a credit into another location managed by OB).

Reply:
```
boolean getIsInternal() { 
   return transactionScope == TransactionScope.INTERNAL
}

TransactionScope getTransactionScope() { 
   return origin == destination ? TransactionScope.INTERNAL : TransactionScope.EXTERNAL
}
```